test/benchmark.js

```diff
@@ -0,0 +1,57 @@
+const { OatyArray } = require('../dist/index.js')
+const testArray = [{
+  a: 1,
+  b: 1,
+  fruit: 'apple'
+}, {
+  a: 1,
+  b: 2,
+  fruit: 'apple'
+}, {
+  a: 1,
+  b: 3,
+  fruit: 'banana'
+}]
+
+var results = []
+
+function time(runs) {
+    var oaty = new OatyArray({data: testArray});
+  
+    var oatyTimes = [];
+    var oatyResults = [];
+    for (var count = 0; count < runs; count++) {
+      const oatyStart = process.hrtime();
+      const oatyResult= oaty.get('fruit', 'banana');
+      const oatyTime = process.hrtime(oatyStart);
+      oatyResults.push(oatyResult);
+      oatyTimes.push(getNanoseconds(oatyTime));
+    }
+  
+    var findTimes = [];
+    var findResults = [];
+    for (var count = 0; count < runs; count++) {
+      const findStart = process.hrtime();
+      const findResult = testArray.find(o => o.fruit === 'banana')
+      const findTime = process.hrtime(findStart);
+      findResults.push([findResult]);
+      findTimes.push(getNanoseconds(findTime));
+    }
+  
+    results.push({runs: runs, oaty: getAverage(oatyTimes), find: getAverage(findTimes)})
+  };
+  
+  function getNanoseconds(hrtime) {
+    return hrtime[0] * 1e9 + hrtime[1];
+  }
+  
+  function getAverage(values) {
+    return values.reduce((previous, current) => current += previous) / values.length;
+  }
+  
+  time(1)

```

Comment:
probably better to use a benchmarking/timing library for this and randomised, variable-length datasets

Reply:
Yea, just wanted something basic for the mean time.
After this is merged we could create a `profiling` branch and make commit all future integration or memory/time profiling tests against that.